src/dotnet/commands/dotnet-restore/NuGetForwardingApp.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Tools.Restore
+{
+    public class NuGetForwardingApp
+    {
+        private const string s_nugetExeName = "NuGet.CommandLine.XPlat.dll";
+        private readonly ForwardingApp _forwardingApp;
+
+        public NuGetForwardingApp(string[] argsToForward)
+        {
+            _forwardingApp = new ForwardingApp(
+                GetNuGetExePath(),
```

Comment:
build scripts duplicate dotnet.deps & dotnet.runtimeconfig for each tool we package with the sdk. This makes them independently runnable so e.g. msbuild doesn't need to understand the workarounds we have in place in CLI.